app/src/main/java/com/example/ithaca_transit_android_v2/NetworkUtils.kt

```diff
@@ -57,4 +59,24 @@ class NetworkUtils {
         val adapter: JsonAdapter<List<Location>> = moshi.adapter(type)
         return adapter.fromJson(body) ?: emptyList()
     }
+
+    fun getRouteOptions(query: String): RouteOptions {
+        val json = JSONObject()
+        json.put("query", query)
+        val requestBody = json.toString().toRequestBody(mediaType)
+        val request: Request = Request.Builder()
+            .url(url + "route")
+            .post(requestBody)
+            .build()
+
+        val body = client.newCall(request).execute().body?.string()
+        val type = newParameterizedType(RouteOptions::class.java)
+        val moshi = Moshi.Builder()
+            .add(RouteAdapter())
```

Comment:
i'm pretty sure that you can throw in a `.add(LocationAdapter())` here and it'll reduce some of the code that you have in your custom adapter. we want to reuse as much code as possible!